Library/Homebrew/cmd/update.sh

```diff
@@ -302,6 +312,9 @@ EOS
   # kill all of subprocess on interrupt
   trap '{ pkill -P $$; wait; exit 130; }' SIGINT
 
+  HOMEBREW_UPDATE_FAILED_FILE="$HOMEBREW_REPOSITORY/.git/UPDATE_FAILED"
```

Comment:
`HOMEBREW_UPDATE_FAILED_FILE` is only used in this function. Let's call it `update_failed_file` and make it local.